fifo.go

```diff
@@ -41,6 +41,11 @@ type fifo struct {
 
 var leakCheckWg *sync.WaitGroup
 
+var ReadWrOnlyFifoError = errors.New("reading from write-only fifo")
```

Comment:
errors shoudl be in the format of `Err*`

So these should be `ErrReadWRONLY`, `ErrReadClosed` `ErrWriteRDONLY` `ErrWriteClosed`